sensors/src/main/scala/com/lynbrookrobotics/potassium/sensors/imu/DigitalGyro.scala

```diff
@@ -0,0 +1,76 @@
+package com.lynbrookrobotics.potassium.sensors.imu
+
+import java.util
+
+import com.lynbrookrobotics.potassium.{PeriodicSignal, Signal}
+import squants.motion.{AngularVelocity, DegreesPerSecond}
+import squants.space.Degrees
+import squants.{Angle, Time}
+
+abstract class DigitalGyro {
+  // Tick Period of the robot
+  final var TICK_PERIOD: Time = ??? //TODO set value based on final 2017 robot
+  val velocityZero = DegreesPerSecond(0)
+  var currentVelocity = Value3D[AngularVelocity](velocityZero, velocityZero, velocityZero)
+  var currentPosition = Value3D[Angle](Degrees(0), Degrees(0), Degrees(0))
+  var currentDrift: Value3D[AngularVelocity] = null
+
+  // List of velocities used for calibration of IMU
+  var calibrationVelocities: util.ArrayList[Value3D[AngularVelocity]] = new util.ArrayList(200)
+
+  // Current index in calibrationVelocities
+  var index: Int = 0
+
+  // Whether IMU is calibrating
+  var calibrating: Boolean = true
+
+  // Gets the current velocity.
+  def retrieveVelocity: Value3D[AngularVelocity]
+
+  // Update velocity stored calibrationVelocities at index
+  def calibrateUpdate(): Unit = {
+    currentVelocity = retrieveVelocity
+    calibrationVelocities.set(index, currentVelocity)
+    index += 1
+
+    if (index >= 200) index = 0
+  }
+
+  // Updates values for the angle on the gyro.
+  def angleUpdate(): Unit = {
+    if (calibrating) {
+      val sum = Value3D[AngularVelocity](velocityZero, velocityZero, velocityZero)
+      calibrationVelocities.forEach(
+        v => sum + v
+      )
+
+      currentDrift = sum.times(-1D / calibrationVelocities.size())
+      calibrationVelocities = null
+      calibrating = false
+    }
+
+    // Stores the value as a form of memory
+    // Modifies velocity according to drift and change in position
+    currentVelocity = retrieveVelocity + currentDrift
+    currentPosition += Value3D[Angle]( Degrees(currentVelocity.valueX * TICK_PERIOD.toSeconds),
```

Comment:
Don't integrate here, instead implement the `position` signal as `velocity.integral`.